src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,668 @@
+    A lot of the features are common with the API interface. In particular, the XRSL
+    language is used in both cases. So, we retain the xrslExtraString and xrslMPExtraString strings.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+__RCSID__ = "$Id$"
+
+import six
+import os
+import stat
+import sys
+
+import requests
+import json
+import ldap3
+
+# from urllib.parse import urljoin
+
+from DIRAC import S_OK, S_ERROR, gConfig
+from DIRAC.ConfigurationSystem.Client.Helpers.Resources import getCESiteMapping
+from DIRAC.Core.Utilities.Subprocess import shellCall
+from DIRAC.Core.Utilities.File import makeGuid
+from DIRAC.Core.Utilities.List import breakListIntoChunks
+from DIRAC.Core.Security.ProxyInfo import getVOfromProxyGroup
+from DIRAC.Resources.Computing.ARCComputingElement import ARCComputingElement
+from DIRAC.Core.Security.ProxyInfo import getProxyInfo
+from DIRAC.Core.Security.X509Chain import X509Chain  # pylint: disable=import-error
+from DIRAC.Core.Security import Locations
+from DIRAC.Core.Utilities.ReturnValues import returnValueOrRaise
+from DIRAC.ConfigurationSystem.Client.Helpers.Path import cfgPath
+
+MANDATORY_PARAMETERS = ["Queue"]  # Mandatory for ARC CEs
+
+# Note : interiting from ARCComputingElement. See https://github.com/DIRACGrid/DIRAC/pull/5330#discussion_r740907255
+class AREXComputingElement(ARCComputingElement):
+
+    # Used in getJobStatus
+    mapStates = {
+        "Accepted": "Scheduled",
+        "Preparing": "Scheduled",
+        "Submitting": "Scheduled",
+        "Queuing": "Scheduled",
+        "Undefined": "Unknown",
+        "Running": "Running",
+        "Finishing": "Running",
+        "Deleted": "Killed",
+        "Killed": "Killed",
+        "Failed": "Failed",
+        "Hold": "Failed",
+        "Finished": "Done",
+        "Done": "Done",
+        "Other": "Done",
+    }
+
+    #############################################################################
+    def __init__(self, ceUniqueID):
+        """Standard constructor."""
+        super(AREXComputingElement, self).__init__(ceUniqueID)
+
+        self.submittedJobs = 0
+        self.mandatoryParameters = MANDATORY_PARAMETERS
+        self.pilotProxy = ""
+        self.queue = ""
+        self.ceHost = self.ceName
+
+        result = returnValueOrRaise(getCESiteMapping(self.ceHost))
+        self.site = result[self.ceHost]
+        self.grid = self.site.split(".")[0]
+
+        # For the REST interface
+        if self.initialised:  # a bunch of the local variables are set up here.
+            self.log.info("REST interface successfully configured", "for %s" % self.ceHost)
+        else:
+            self.log.info("REST interface not yet properly configured", "for %s" % self.ceHost)
+            self.log.info("Try again in next iteration", "Bye Bye")
+            return S_ERROR("Try try try initialisation again")
```

Comment:
You can't return a result from `__init__`. The only option is to raise an exception:

```suggestion
            raise RuntimeError("REST interface not yet properly configured")
```